app/Http/Controllers/StudentOperation.php

```diff
@@ -349,6 +370,50 @@ public function confirm_session($user_id, Request $request)
         }
     }
 
+    //Display change course form
+
+    public function change_course()
+    {
+        $user = Auth::user();
+        $courses = Course::all();
+        
+        // Current course (if any)
+        $currentCourse = null;
+        if (!empty($user->exam)) {
+            $currentCourse = Course::find($user->exam);
+        }
+        
+        return view('student.change-course', compact('user', 'courses', 'currentCourse'));
+    }
+    
+
+   // Update course selection
+
+   public function update_course(Request $request)
+{
+    $request->validate([
+        'course_id' => 'required|exists:courses,id',
+        'session' => 'required'
```

Comment:
only update course not session, student is not admitted at this point so not session to choose